### PR TITLE
Updated Assessment...

### PR DESCRIPTION
**Pull request by umairarshad47**
1. Build runs successfully
2. No errors in browser console. No lint issues.
3. Project structure follows atomic design (as per docs)
4. You may skip the landing page and sidebar/drawer.
5. Modal with steps is implemented with pixel-perfect UI and functionality.
6. Components are not very coupled for the sake of project scaleability.
7. Loom Video: https://www.loom.com/share/267ef241e5e04ac7abc4ca55612cffce?sid=2cd0d6c1-98dd-4f2f-85ca-20412850d782

### CODE REVIEW

File: src/organisms/modals/assign-journey-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                                variant="outlinePurple"
+                                label="Go to my Journeys"
+                                onClick={onClose}
+                            />
+                        </div>
+                    )}
+
+                {selectedUser.length < 1 && (
+                    <div className="flex flex-col justify-center items-center mt-3">
+                        <div className="text-[12px]">Or use <span className="underline text-[#8b46f6]">learner's lab</span> to assign custom groups of learners</div>
```
Comment (uxaircodes) [inline src/organisms/modals/assign-journey-modal.tsx:183]: Try to avoid using the hard-coded css values and try to add primary colors like this one in tailwind config and utilize that instead.

File: src/app-components/user-selection.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            </div>
+        </components.Option>
+    );
+};
+
+const UserSelection = ({ users, selectedUser, onChange, error }: UserSelectionProps) => {
+    const [inputValue, setInputValue] = useState<string>("");
+    return (
+        <>
+            <Select
```
Comment (uxaircodes) [inline src/app-components/user-selection.tsx:64]: Strictly follow the UI requirements from Figma file. As right now the menu is not matching with the UI at all.

File: src/organisms/modals/assign-journey-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                            selectedUser={selectedUser}
+                            onChange={(val: UserOption[]) => {
+                                setErr(null);
+                                setSelectedUser(val);
+                            }}
+                            error={err}
+                        />
+                    )}
+                    {step === 2 && <NotificationSettings />}
+                    {step === 3 && <SuccessMessage />}
```
Comment (uxaircodes) [inline src/organisms/modals/assign-journey-modal.tsx:153]: No confetti animation on success modal. Also the width does not align with what was required in Figma.

**Comment by uxaircodes** [comment]
There are linting errors that need to be addressed. In our production pipelines, the build does not pass unless all lint issues are resolved. Please ensure that both the build and lint checks are clear before submitting a PR for review.
Screenshot 2025-07-01 at 11 13 34 AM

The multi-select menu is currently not displaying any user list, which creates the impression that the admin must remember and type each name manually. However, the Figma design clearly indicates that users should be listed in the dropdown, ideally with 50 records per request to avoid overwhelming the UI with hundreds of entries.
Screenshot 2025-07-01 at 11 16 02 AM

**Comment by uxaircodes** [comment]
**PR Submission Guidelines**
Please also ensure that once you’ve completed the implementation of your assigned task, you perform a thorough QA pass before submitting the PR. This helps surface any overlooked issues, validates that the feature works as expected, and significantly improves the overall stability and readiness of the code for production.

**Additional best practices to follow before PR submission:**
- Run and pass all lint and build checks. Ensure no linting or compilation errors exist.
- Conduct end-to-end and edge case testing. Validate the flow across different scenarios, including unexpected or extreme cases.
- Cross-check with Figma designs. UI/UX should align with the design system and visual expectations.
- Test responsiveness. Make sure the UI works across various screen sizes if applicable.
- Self-review your code. Read through your own changes to catch any mistakes, unused code, or opportunities to improve readability.
- Verify integration points. Check API calls, state management, and any shared components for potential breakage.
- Clean up console logs and unused code. Avoid leaving debug statements or redundant code in your PR.

By following these practices, we ensure higher-quality deliveries, reduce review time, and promote smoother merges into the main branch.

File: src/organisms/modals/assign-journey-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+            <DialogContent className="bg-white max-h-[calc(100vh-10rem)] overflow-y-auto">
+                <DialogHeader>
+                    <DialogTitle
+                        className="flex items-center gap-2 justify-center text-2xl"
+                        style={{ fontFamily: 'Georgia, "Times New Roman", Times, serif' }}
+                    >
+                        {step !== 1 && step !== 3 && (
+                            <ChevronLeftCircleIcon className="cursor-pointer" onClick={handleBackStep} />
+                        )}
+                        {stepBasedHeaders(step)}
```
Comment (uxaircodes) [inline src/organisms/modals/assign-journey-modal.tsx:103]: It is not wise to use steps like how we have implemented here. Instead we should isolate each step in separate component and for rendering we can utilize switch statements for these components. It will make the component more organized and readable.
